[2 of n] DCOS-6523: Add config modal

jfurrow opened the pull request:
Depends on #129

mesosphere-web commented:
Reviewer: @aldipower

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+    );
+  }
+
+  getModalContents() {
```
can we put this method right above `render`?

jfurrow replied:
Sure, but what is the logic for this? I guess I'm just not sure how to order these methods

kennyt replied:
well usually when i review, i go to `render` and then if there's a main rendering function inside there, i try to look for it.

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+        modalClass={modalClasses}
+        modalWrapperClass="multiple-form-modal"
+        onClose={this.handleModalClose}
+        open={this.props.open}
```
i see we use `props.open` but don't have it in the proptypes

jfurrow replied:
Done!

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+}
+
+UpdateConfigModal.propTypes = {
+  service: React.PropTypes.instanceOf(Service).isRequired
```
somewhere in this component we have `props.servicePackage` and her it's `service`. what is it?

jfurrow replied:
It's both! `service` is the instance of `Framework` and `servicePackage` is an instance of `UniversePackage`, I'm just following the naming convention from the `Install` modal but it would make sense to name it `framework` and `package` :)

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+      updateError: null,
+      updateSuccess: false
+    });
+    this.props.onClose();
```
this needs to be in proptypes

jfurrow replied:
Done!

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+    // current configuration.
+    // JIRA Task DCOS-6638
+    let currentConfiguration = {};
+    let {name, version} = service.get('package');
```
`service.getPackage()`

jfurrow replied:
Done!

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+    let currentConfiguration = {};
+    let {name, version} = service.get('package');
+    let icon = service.getIcons()['icon-small'];
+    let schema = service.get('config');
```
there must be a `getConfig`

jfurrow replied:
There's not, but I added one!

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+      <div className="horizontal-center modal-content-inner container
+        container-pod container-pod-short text-align-center">
+        <h2 className="flush-top short-bottom">Success!</h2>
+        <div style={{width: '100%', overflow: 'auto', wordWrap: 'break-word'}}
```
any reason why we need to use `style` tags?

jfurrow replied:
Just sloppy copied from the install modal, I'll make a class for it.

jfurrow replied:
Done!

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+    );
+  }
+}
+
```
let's get some defaultprops up in this

jfurrow replied:
Done!

kennyt commented on src/js/components/modals/UpdateConfigModal.js:
```diff
@@ -0,0 +1,255 @@
+  getUpdateSuccessContent() {
+    let {servicePackage} = this.props;
+
+    let notes = Util.findNestedPropertyInObject(
```
we don't have a `.getPostInstallNotes`?

jfurrow replied:
Hah, we do! Done!

kennyt commented:
let's rebase this since #1 is merged in 👍

jfurrow commented:
@kennyt rrrrrrrrrrrebased

kennyt commented on src/js/components/ServiceOptions.js:
```diff
@@ -63,14 +64,19 @@ class ServiceOptions extends mixin(StoreMixin) {
       <a className={editButtonClasses} key="edit-config"
         onClick={this.handleEditConfigClick}>
         Edit
+        <UpdateConfigModal open={state.editConfigModalOpen}
```
can we have one prop per line? :)

jfurrow replied:
Did we decide to do this consistently at some point? TBH I'm not a fan of one prop per line, but if that's what we decided then I'll stick to it

jfurrow replied:
Done!

kennyt commented:
currently the modal doesn't animate in for me and also, the `Back` button in the footer doesn't close it

kennyt commented:
tests failing

jfurrow commented:
Looks like I accidentally pushed a bunch of stuff I didn't mean to... :(

jfurrow commented:
@kennyt I removed the mistakenly committed code, fixed the "back" button, and fixed the modal's animation in #138 so this should be ready for review again :)
